refactor(toolkit-lib): diff action uses helper methods

kaizencc opened the pull request:
refactors the diff action to use helper methods and in general cleans up the code a lot. introduces an internal concept of `TemplateInfo` that standardizes the to-be-diffed information, so there's not a bunch of repetitive code

existing tests continue to succeed. 

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license

kaizencc committed:
refactor(toolkit-lib): diff action uses helper methods

kaizencc committed:
improts

kaizencc committed:
Merge branch 'main' into conroy/refactor-diff

mrgrain commented on packages/@aws-cdk/toolkit-lib/lib/actions/diff/index.ts:
this seems to all be private helpers, so they should go into `diff/private/helpers.ts` or something

kaizencc committed:
move functions around

kaizencc committed:
extraneous code

kaizencc committed:
imports

mrgrain commented on packages/@aws-cdk/toolkit-lib/lib/actions/diff/index.ts:
```diff
@@ -1,4 +1,4 @@
-import type { StackSelector } from '../../api/cloud-assembly';
+import type { StackSelector } from '@aws-cdk/tmp-toolkit-helpers';
```
or whatever the path is
```suggestion
import type { StackSelector } from '../../@aws-cdk/tmp-toolkit-helpers';
```

mrgrain commented on packages/@aws-cdk/toolkit-lib/lib/toolkit/toolkit.ts:
```diff
@@ -268,122 +266,40 @@ export class Toolkit extends CloudAssemblySourceBuilder implements AsyncDisposab
 
-    if (diffMethod.method === 'local-file') {
-      const methodOptions = diffMethod.options as LocalFileDiffOptions;
+    const templateInfos = await makeTemplateInfos(ioHelper, stacks, deployments, await this.sdkProvider('diff'), options);
```
So this is preparing the diff? Current names read unclear to me. Maybe something like this is clearer:

```suggestion
    const diffInput = await prepareDiff(ioHelper, stacks, deployments, await this.sdkProvider('diff'), options);
```

kaizencc replied:
i like the prepareDiff name but also after my latest refactor i think the type of `TemplateInfo` now makes more sense -- since its only getting passed into `DiffFormatter`. lemme know if you disagree

Commit:
chore: self mutation

Signed-off-by: github-actions <github-actions@github.com>

kaizencc committed:
merge

kaizencc committed:
pr feedback

kaizencc committed:
fix tests

Commit:
chore: self mutation

Signed-off-by: github-actions <github-actions@github.com>

kaizencc committed:
Merge branch 'main' into conroy/refactor-diff

kaizencc committed:
Merge branch 'main' into conroy/refactor-diff

mrgrain commented on packages/@aws-cdk/tmp-toolkit-helpers/test/api/diff/diff.test.ts:
```diff
@@ -59,20 +59,18 @@ describe('formatStackDiff', () => {
+          templateFile: 'template.json',
+          stackName: 'test-stack',
+          findMetadataByType: () => [],
+        } as any,
```
there are proper helpers for this around

mrgrain replied:
why not `mockNewTemplate`?

kaizencc replied:
oops

mrgrain commented on packages/@aws-cdk/toolkit-lib/lib/actions/diff/index.ts:
```diff
@@ -1,4 +1,4 @@
-import type { StackSelector } from '../../api/cloud-assembly';
+import type { StackSelector } from '../../../lib/api/shared-public';
```
Any reason for this?

kaizencc committed:
Merge branch 'main' into conroy/refactor-diff

kaizencc committed:
pr feedback

kaizencc committed:
one line

kaizencc committed:
Merge branch 'main' into conroy/refactor-diff

kaizencc committed:
Merge branch 'main' into conroy/refactor-diff

mrgrain committed:
refactor(cli): inject CLI version as a parameter to Notices (#323)

`Notices` will eventually serve both toolkit-lib and cli. Currently the
class has a hard-coded "cli version" - but really the value is reading
the version from whatever the current package is. Once we start using
this file in another package, it will be wrong. For now, let's just
remove the hard dependency on cli version inside the class. In future it
will be extended.

- Remove direct dependency on versionNumber import in Notices class
- Add cliVersion as a required parameter in NoticesProps interface
- Store cliVersion as a class property and use it in display method
- Improve testability by removing the need to mock version functions

---
By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache-2.0 license

mrgrain committed:
refactor(toolkit-lib): move payloads out of api (#325)

The message payload interfaces never were an api. They just ended up
there by chance. Move them to their own top-level module. This has the
side-effect that it will make future diffs slightly nicer.

---
By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache-2.0 license

mrgrain committed:
refactor(cli): rename `logs` to `logs-monitor` (#324)

We are about to move this module into a different package, and the
target package is gitignoring any directory called `logs`. We could just
fix the gitignore, but the module name is very inacurate anyway.
Therefore we are renaming it.

Also moves `rwlock` from `api/utils` to `api`. This was an oversight and
doesn't warrant a separate PR.

Finally moves some test files into their correct corresponding
directories.

---
By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache-2.0 license

iliapolo committed:
chore: enable atmosphere (#316)

Once this is merged we can increase concurrency. 

---
By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache-2.0 license

mrgrain committed:
fixup

mrgrain committed:
Merge remote-tracking branch 'origin/main' into conroy/refactor-diff
